### DISCUSSION TITLE
How to deactivate unusable bands?

### DISCUSSION THREAD

**Discussion by R1BLH** [magicbug/Cloudlog#1591] [Ideas / Feature Requests]
It is possibility to deactivate some unusable modes in the interface. 
It would be nice for the same possibility for unusable bands.

**Comment by dg9vh** [comment]
I second this - but I think this would only be of sense for the logging-window and not for editing and import/export of existing adif-data. It could be possible that you import an adif (for example created by wsjt-x) where you worked on Bands you usually not operate in SSB/CW (where you log directly in cloudlog, for example).

**Comment by AndreasK79** [comment]
Then I say we need something like this:
![image](https://user-images.githubusercontent.com/6977712/188286138-1f5c9108-f476-4cac-a8bd-8e0b4bb92eda.png)
Bands and usersettings should be stored in the database, and code updated to use the bands that are active. For awards, only the bands checked.

**Reply by dg9vh** [reply]
looks good to me...

**Reply by AndreasK79** [reply]
That was just a quick mockup. It does nothing, yet...

My thoughts:

1. A table for holding bands (band table)
2. A table holding the band information for the user (settings from the picture above) (bandxuser table)
3. Migration script to create the tables and fill
4. Fill the table (bandxuser) when creating a user
5. A view like the one above, to edit settings, create/edit/delete bands
6. Implement the band fetching all places needed (awards/qso editing/analytics?)

I think that should cover the bases. Ideas welcome.

**Reply by magicbug** [reply]
Sorry been out in a field all weekend, main thing is this is user level not global.

**Reply by AndreasK79** [reply]
Yeah, that's the idea. I'd say that I'm 40-50% done on this.

**Comment by AndreasK79** [comment]
PR #1595 is available now for those willing to test. User selectable bands can now be set on a user level. Feedback welcome.

**Comment by AndreasK79** [comment]
This is merged into station_logbooks branch.
